packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { authentication, FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppAuthentication {
+  /*********** BEGIN REQUEST TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type OriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionRequest {
+    requestType: OriginalRequestType.QueryMessageExtensionRequest;
+    commandId: string;
+    parameters?: {
+      name?: string;
+      value?: string;
+    }[];
```

Comment:
The current definition allows for all these:
```
interface IFoo {
    parameters?: {
      name?: string;
      value?: string;
    }[];
}

const myFoo1: IFoo = { }
const myFoo2: IFoo = { parameters: undefined }
const myFoo3: IFoo = { parameters: [] }
const myFoo4: IFoo = { parameters: [{}] }
const myFoo5: IFoo = { parameters: [{name: undefined}] }
const myFoo6: IFoo = { parameters: [{name: undefined, value: undefined}] }
```

Do each of those actually have semantically different meanings that we want to allow them? Or are some/all of them representing the same state and we're making more work for ourselves and others to have to infer the same semantic state from multiple possible values?

I'm cheating a little bit because `myFoo1` and `myFoo2` are really just syntactic sugar for the same thing. There's no way to allow `myFoo1` and prevent `myFoo2` but that's okay, they are both checked the same way. This is also true for `myFoo4` which is equivalent and not distinct from `myFoo5` and `myFoo6`. Really this is the list:

```
const myFoo1: IFoo = { }
const myFoo3: IFoo = { parameters: [] }
const myFoo4: IFoo = { parameters: [{}] }
```

Is it really correct that both `name` and `value` could be undefined? What would it mean for a `parameters` element to exist but have no member properties? And how would that semantically be different than just not having a parameters entry at all? What is the semantic difference between having a parameters array with zero elements (which is allowed by this definition) and a parameters array that was `undefined`?

Reply:
I agree that more information is needed here. Due to timelines, I'd like to take this as a follow-up item.

To make progress on this, @ashishsingh2105 and @susanshen96 -- is it possible to get more specific information about the valid combinations of values that can be provided?

Reply:
We can keep it to be:
`parameters: {
    name: string;
    value: string;
}`

Reply:
But still an array? E.g., `parameters: { name: string; value: string; }[];`